### PR TITLE
Support unstable features and custom target

### LINKED ISSUE #62: Allow use with Cargo's "build-std" feature
Cargo has a recent nightly-only feature for building the Rust standard library (std) as part of a normal build: https://github.com/rust-lang/wg-cargo-std-aware

Consider this example project:
```sh
cargo new bloattest
cd bloattest
cat <<'EOF' >> Cargo.toml

[profile.release]
opt-level = "z"
lto = true
codegen-units = 1
panic = "abort"
EOF
cargo build --release --target x86_64-unknown-linux-gnu
```

`cargo-bloat` works properly here:

```sh
cargo bloat --release --target x86_64-unknown-linux-gnu
```

It generates this output:
```
    Finished release [optimized] target(s) in 0.00s
    Analyzing target/x86_64-unknown-linux-gnu/release/bloattest

 File  .text     Size     Crate Name
 2.2%  16.7%  28.7KiB       std std::backtrace_rs::symbolize::gimli::elf::<impl std::backtrace_rs::symbolize::gimli::Mapping>::new
 0.8%   6.2%  10.7KiB       std addr2line::ResUnit<R>::parse_lines
 0.7%   5.3%   9.2KiB       std miniz_oxide::inflate::core::decompress
 0.6%   4.8%   8.3KiB       std std::sys_common::backtrace::_print_fmt::{{closure}}
 0.4%   2.8%   4.9KiB       std addr2line::ResUnit<R>::parse_functions
 0.4%   2.8%   4.8KiB       std gimli::read::unit::parse_attribute
 0.3%   2.4%   4.1KiB       std rustc_demangle::v0::Printer::print_type
 0.3%   2.3%   3.9KiB       std addr2line::Function<R>::parse_children
 0.3%   2.2%   3.7KiB       std <std::backtrace_rs::symbolize::SymbolName as core::fmt::Display>::fmt
 0.3%   2.1%   3.6KiB       std gimli::read::rnglists::RngListIter<R>::next
 0.3%   1.9%   3.2KiB       std core::slice::sort::recurse
 0.2%   1.8%   3.1KiB       std rustc_demangle::try_demangle
 0.2%   1.7%   2.9KiB [Unknown] main
 0.2%   1.3%   2.3KiB       std gimli::read::line::parse_attribute
 0.2%   1.3%   2.2KiB       std std::sys_common::backtrace::output_filename
 0.2%   1.2%   2.2KiB       std core::str::pattern::StrSearcher::new
 0.2%   1.2%   2.1KiB       std rustc_demangle::v0::Printer::print_path
 0.1%   1.1%   1.9KiB       std gimli::read::unit::Attribute<R>::value
 0.1%   1.0%   1.7KiB       std std::backtrace_rs::symbolize::gimli::callback
 0.1%   0.9%   1.6KiB       std std::backtrace_rs::print::BacktraceFrameFmt::print_raw
 5.0%  37.8%  65.0KiB           And 228 smaller methods. Use -n N to show more.
13.4% 100.0% 172.1KiB           .text section size, the file size is 1.3MiB
```

Cargo's new `build-std` feature can reduce the executable size:
```sh
rustup component add rust-src

cargo build --release \
    --target x86_64-unknown-linux-gnu \
    -Z build-std=panic_abort,std \
    -Z build-std-features=panic_immediate_abort
```

`cargo-bloat` doesn't work with this build method:
```sh
cargo bloat --release \
    --target x86_64-unknown-linux-gnu \
    -Z build-std=panic_abort,std \
    -Z build-std-features=panic_immediate_abort
```

The above results in this error:
```
Error: unused arguments left: -Z, build-std=panic_abort,std, -Z, build-std-features=panic_immediate_abort.
```

If `cargo-bloat` would pass along the various nightly-only `-Z` arguments to `cargo build`, it could be used successfully.  As a proof-of-concept, here is a `cargo` shim to force the `build-std` arguments:

```sh
mkdir cargo-bin
cat <<'EOF' > cargo-bin/cargo
#!/bin/sh

if [ "$1" = "build" ]; then
    shift
    set -- build \
        -Z build-std=panic_abort,std \
        -Z build-std-features=panic_immediate_abort \
        "$@"
fi

exec ~/.rustup/toolchains/nightly-2020-08-20-x86_64-unknown-linux-gnu/bin/cargo "$@"
EOF
chmod +x cargo-bin/cargo
```

The shim assumes the toolchain "nightly-2020-08-20" for convenience.  Now `cargo-bloat` works properly:

```sh
PATH="$PWD/cargo-bin:$PATH" cargo bloat --release \
  --target x86_64-unknown-linux-gnu
```

It generates this output:
```
    Finished release [optimized] target(s) in 0.06s
    Analyzing target/x86_64-unknown-linux-gnu/release/bloattest

 File  .text    Size     Crate Name
 3.1%   9.6%  1.4KiB [Unknown] main
 2.5%   7.9%  1.1KiB      core <str as core::fmt::Debug>::fmt
 1.4%   4.3%    631B      std? <std::io::Write::write_fmt::Adaptor<T> as core::fmt::Write>::write_str
 1.4%   4.3%    624B       std std::thread::Thread::new
 1.3%   4.2%    609B       std std::io::stdio::stdout
 1.3%   3.9%    567B bloattest bloattest::main
 1.2%   3.8%    554B      core core::fmt::Formatter::pad_integral
 1.1%   3.3%    478B       std std::sync::once::Once::call_once::{{closure}}
 1.0%   3.1%    454B      core core::fmt::write
 0.7%   2.2%    319B       std std::sys::unix::thread_local_dtor::register_dtor
 0.7%   2.2%    315B       std std::io::buffered::BufWriter<W>::flush_buf
 0.6%   2.0%    290B       std std::sys::unix::stack_overflow::imp::signal_handler
 0.6%   1.9%    282B      core core::fmt::Formatter::pad
 0.5%   1.5%    216B      core core::unicode::printable::check
 0.4%   1.4%    199B      std? <std::io::Write::write_fmt::Adaptor<T> as core::fmt::Write>::write_str
 0.4%   1.3%    196B       std <std::io::buffered::BufWriter<W> as std::io::Write>::write
 0.4%   1.3%    192B      core core::fmt::num::imp::fmt_u64
 0.4%   1.2%    182B      core core::str::slice_error_fail
 0.4%   1.2%    178B       std std::sys_common::thread_info::THREAD_INFO::__getit
 0.4%   1.2%    173B       std <std::io::stdio::Stdout as std::io::Write>::write_fmt
11.7%  36.4%  5.2KiB           And 113 smaller methods. Use -n N to show more.
32.3% 100.0% 14.2KiB           .text section size, the file size is 44.1KiB
```

Would it be possible to add support for Cargo's `build-std` feature to `cargo-bloat`?

### PR DESCRIPTION
**Pull request by nook1208**
First of all, thank you for this great feature.
Please confirm this patch to support [unstable features](https://doc.rust-lang.org/cargo/reference/unstable.html) and [custom target](https://docs.rust-embedded.org/embedonomicon/custom-target.html).
It can fixes #62 issue also.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -281,6 +283,7 @@ pub struct Args {
     bin: Option<String>,
     example: Option<String>,
     test: Option<String>,
+    z: Option<String>,
```
Comment (RazrFalcon) [inline src/main.rs:286]: It has to be `Vec<String>`

File: src/main.rs [hunk 2]
```diff
@@ -289,6 +292,7 @@ pub struct Args {
     no_default_features: bool,
     target: Option<String>,
     target_dir: Option<String>,
+    rustlib_target: Option<String>,
```
Comment (RazrFalcon) [inline src/main.rs:295]: What is this flag?

File: src/main.rs [hunk 3]
```diff
@@ -312,6 +316,7 @@ fn parse_args(raw_args: Vec<std::ffi::OsString>) -> Result<Args, pico_args::Erro
         bin:                    input.opt_value_from_str("--bin")?,
         example:                input.opt_value_from_str("--example")?,
         test:                   input.opt_value_from_str("--test")?,
+        z:                      input.opt_value_from_str("-Z")?,
```
Comment (RazrFalcon) [inline src/main.rs:319]: It has to be `input.values_from_str("-Z")`

**Review by RazrFalcon — CHANGES_REQUESTED** [review]

File: src/main.rs (hunk 2 above)
Comment (nook1208) [reply src/main.rs:295]: If we use custom target, we use `--target` like this : `--target foo.json`.
And we can specify the [llvm-target](https://docs.rust-embedded.org/embedonomicon/custom-target.html#fill-the-target-file) in json.
So, the rustlib_target means llvm-target. (e.g. aarch64-unknown-none)

File: src/main.rs (hunk 1 above)
Comment (nook1208) [reply src/main.rs:286]: Ok, I'll test it.

File: src/main.rs (hunk 3 above)
Comment (nook1208) [reply src/main.rs:319]: Ok, I'll test it.

File: src/main.rs (hunk 2 above)
Comment (RazrFalcon) [reply src/main.rs:295]: Does cargo itself has this flag?

Comment (nook1208) [reply src/main.rs:295]: I guess not.

Comment (RazrFalcon) [reply src/main.rs:295]: Then we have to skip it. `cargo bloat` mimics `cargo build` commands.

Comment (nook1208) [reply src/main.rs:295]: I see. I'll find another way then. 
Thank you for the review.

**Pull request closed without merging**